active-rfcs/0004-api-versioning.md

```diff
@@ -0,0 +1,65 @@
+- Start Date: 2020-05-28
+- Target Major Version: 2.0.0
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+In order to improve our legacy API and code, we want to add or change some functionality that may cause client versions that consume our API to stop working. To solve this problem and add more functionalities without breaking anything, making the new clients (apps and websites) consume the new functionalities, we must add and manage different versions of our API.
+
+
+# Basic example
+
+```bash
+curl -i -H " Accept: application/vnd.wawmuzik.v1+json" -H "Content-Type: application/json" http://app.wawmuzik.com/auth/login
+```
+
+and the API will send a response with this header:
+
+```bash
+Content-Type: application/vnd.wawmuzik.v1+json
+```
+
+# Motivation
+
+We want to improve our API and business logic adding more semantic data and improving a lot of things, for no break anything semantic versioning will the best approach for this. For now, we will only mayor versions, so all clients will be using something like v1, v2, v3 ...
+
+# Detailed design
+
+We will manage our API through semantic versioning, being the client who defines which version of the API they want to consume using this header
+
+```bash
+curl -i -H " Accept: application/vnd.wawmuzik.v1+json" -H "Content-Type: application/json" http://app.wawmuzik.com/auth/login
+```
+
+the response will be come with this type of header:
+
+```bash
+Content-Type: application/vnd.wawmuzik.v1+json
+```
+
+In backend, in our routers, we will need to check this header before do something in the ednpoint, so we will need to make a Factory that manage this behaviour.
+
+For now, requests that doesn't have this header, will be redirect to the first version of our API. This approach will be for now, because in the future, when we will remove this 1 version, requests that doesn't have this header, will be redirect to the lastest version
```

Comment:
```suggestion
For now, requests that don't have this header will be redirected to the first version of our API. This approach will be for now, because in the future, when we will remove this 1 version, requests that don't have this header, will be redirected to the latest version
```